packages/nuxi/src/cli-wrapper.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * This file is used to wrap the CLI entrypoint in a restartable process.
+ */
+import { fileURLToPath } from 'node:url'
+import { execa } from 'execa'
+import { EXIT_CODE_RESTART } from './constants'
+
+const cliEntry = fileURLToPath(new URL('../dist/cli-run.mjs', import.meta.url))
+
+async function startSubprocess (preArgs: string[], postArgs: string[]) {
+  const child = await execa(
```

Comment:
Perf: I guess we can use `fork` from `nod:child_process`. This is also consistent with the final nuxi-ng implementation and provides a communication channel.

Reply:
I see we use `execa` in many places already, and the chunk is shared. I guess the performance won't be a big issue here.

I am just worried about using `fork` or `exec` might create more issues to handle on different OS, etc.

Reply:
We could use 1 btw too. Auto restarting with child unwanted exit codes (also consistent with sighup used for restart)

Reply:
Main usecase i added execa was search path compatibility which isn’t a case here. Also forke copies mem in unix then forks which by theory at least should be faster than loading node to the mem again.

Reply:
Could we mesure the performance difference here?

I like the simplicity of code and execa has some good features:
<img width="835" alt="CleanShot 2023-03-03 at 10 12 41@2x" src="https://user-images.githubusercontent.com/904724/222680167-34383749-6b2f-49e8-9675-a73a0b46352a.png">

Reply:
Most of those benefits aren't relevant here and main benefit is using fork over exec with fine gained control. we are forking a known entry as process (not a on-off executation).

Finally this impl will be part of citty's dev server so this change is to make sure we are aligned :)